src/components/nodes/pool/pool.vue

```diff
@@ -173,6 +173,9 @@ export default {
 
       await Promise.all(lanes);
       this.$emit('set-shape-stacking', this.shape);
+      this.graph.getLinks().forEach(link => {
+        this.$emit('set-shape-stacking', link);
+      });
```

Comment:
Based on my previous comment, and our discussion, since this has been added here, the `bringFlowsToFront` function and the `graph` parameter of `ensureShapeIsNotCovered` should be removed. This address @JCook21's comments as well.